packages/ember-glimmer/lib/environment.js

```diff
@@ -124,7 +125,7 @@ export default class Environment extends GlimmerEnvironment {
       templates
     } = statement;
 
-    if (isSimple && (isInline || isBlock)) {
+    if (key !== 'partial' && isSimple && (isInline || isBlock)) {
```

Comment:
with short circuiting you want the common cases first, check the booleans before doing the key compare